pkg/server/instrumentation_service_test.go

```diff
@@ -0,0 +1,37 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRunInstrumentationService(t *testing.T) {
+	s, err := NewInstrumentationService(log.New("test-logger"))
+	require.NoError(t, err)
+
+	err = s.start(context.Background())
+	require.NoError(t, err)
+
+	go func() {
+		_ = s.running(context.Background())
+	}()
+
+	require.NoError(t, err)
```

Comment:
```suggestion
	ctx, cancel := context.WithDeadline(context.Background(),30)
	defer cancel()
	
	err = s.StartAsync(ctx)
	require.NoError(t, err)

	err = s.AwaitRunning(ctx)
	require.NoError(t, err)
```

Reply:
You can also use: [services.StartAndAwaitRunning](https://pkg.go.dev/github.com/grafana/dskit/services#StartAndAwaitRunning) / [StopAndAwaitTerminated](https://pkg.go.dev/github.com/grafana/dskit/services#StopAndAwaitTerminated) as a quick shortcut.